salt/utils/master.py

```diff
@@ -126,6 +128,8 @@ def _get_cached_minion_data(self, *minion_ids):
             log.debug('Skipping cached data because minion_data_cache is not '
                       'enabled.')
             return grains, pillars
+        if not minion_ids:
```

Comment:
@UtahDave, this has the unfortunate effect of targeting everything in the case of a failed target.

i.e.: 
tgt=thingThatNoLongerExists
action=killStuff

Is really dangerous.

Reply:
If the tgt doesn't match it returns all minions..